### DISCUSSION TITLE
Is there any snippet how to create user via Recipe

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#7386] [Q&A] [answered]
Is there any snippet how to create user via Recipe?
As well, is it possible to **disable** user by code?

**Comment by hishamco** [comment, marked as answer]
AFAIK the users will be added through setup or the users page only, I don't think we already have a recipe on the built-in themes for doing such thing, @agriffard  do we have this feature?

/cc @deanmarcussen

**Reply by jtkech** [reply]
I already used it but don't remember all details, i know that @jptissot also used it ;)

That said, a recipe can execute commands through a command step, and there are user commands.

Look at `CommandStep.cs` to see e.g. that the recipe step name is `"Command"` and so on

Look at `UserCommands.cs` to see the available user commands and their syntax.

**Reply by hishamco** [reply]
If through command make sense, I though there's a direct way

**Reply by vitalybrandes** [reply]
Is there any simple way to disable user programmable?

**Reply by Skrypt** [reply]
As of now, I think you can create a User from recipes but not as "disabled" because of the params allowed on the command.

**Reply by vitalybrandes** [reply]
@Skrypt  thank you for your time, 
Actually i need to "disable" OC user remotely. following some event in external application, i need to run local task in OC.
At first point i would like to keep user disabled , but after 30 days totally remove it from OC.
I am wondering is its possible to make specific user disabled, by code.

**Reply by hishamco** [reply]
You can do this programmatically,  perhaps background task may suitable to execute jobs periodically

Please check the Users module you will see how the user is disabled or deleted

**Reply by vitalybrandes** [reply]
Thank you  ;-):

 var user = await _userManager.FindByNameAsync(userName);
 if (!(user as User).IsEnabled)
            {
                reportError(string.Empty, S["The specified user is not allowed to sign in."]);

                return null;
            }
